src/cryptonote_core/miner.cpp

```diff
@@ -365,6 +398,11 @@ namespace cryptonote
         misc_utils::sleep_no_w(100);
         continue;
       }
+      else if( m_is_background_mining_enabled )
+      {
+        misc_utils::sleep_no_w(m_miner_extra_sleep);
```

Comment:
Shouldn't this sleep be skipped when mining? And Is this variable even necessary? Seems like it should be sleeping at the constant `BACKGROUND_MINING_MINER_MONITOR_INVERVAL` rate.

Also what if this were:
```c++
bool is_checking_idle_usage = false;

...
else if (m_is_background_mining_enabled) {
    if (!m_is_background_mining_started) {
        const boost::lock_guard<boost::mutex> lock{m_background_mining};
        if (m_stop)
            break;

        if (!m_is_background_mining_started) {
            wait_for_idle();
           if (m_stop)
               break;

            is_checking_idle_usage = true;
            m_is_background_mining_started = true;
        }
    }
}
...

if (is_checking_idle_usage) {
    if (not_idle()) {
        is_checking_idle_usage = false;
        m_is_background_mining_started = false;
    }
}
```

So all worker threads would block at the mutex while _one of_ the worker threads checked for idle usage. Once it succeeded, it would be the sole thread to check for the stopping condition. The motivation would be to prevent an additional thread from being created, with the downside that one mining thread has to do a few extra system calls after each mining attempt. Perhaps the check could be done every couple attempts, if the system calls were a little expensive.